.github/workflows/benchmarking.yml

```diff
@@ -186,31 +186,101 @@ jobs:
 
       - name: Force Cancel SauceLabs Job on Workflow Cancellation
         if: ${{ cancelled() }}
+        uses: actions/github-script@v7
         env:
           SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
           SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
-        run: |
-          echo "::warning Workflow was cancelled, attempting to cancel SauceLabs jobs"
-
-          # Function to extract and cancel job from log file
-          cancel_jobs_from_log() {
-            local log_file="$1"
-            if [[ -f "$log_file" ]]; then
-              echo "Checking $log_file for SauceLabs test IDs..."
-              # Extract all SauceLabs test IDs from the log
-              grep -oE 'https://app\.saucelabs\.com/tests/[^\s]+' "$log_file" | while read -r url; do
-                test_id=$(echo "$url" | sed 's|.*tests/||')
-                if [[ -n "$test_id" ]]; then
-                  echo "Attempting to cancel SauceLabs job: $test_id"
-                  # Cancel the job using saucectl
-                  saucectl jobs stop "$test_id" || echo "Failed to cancel job $test_id (may already be finished)"
-                fi
-              done
-            fi
-          }
-
-          # Cancel jobs from both possible log files
-          cancel_jobs_from_log "output.log"
-          cancel_jobs_from_log "retry-output.log"
-
-          echo "SauceLabs job cancellation attempts completed"
+        with:
+          script: |
+            const fs = require('fs');
```

Comment:
`m`: Wouldn't it be better to extract this into the `scripts/` folder, so you can also run it manually?